enumerables.rb

```diff
@@ -0,0 +1,199 @@
+        array.push(self[i])
+      end
+      i += 1
+    end
+    array
+  end
+
+  def my_all?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield(self[i]) == false
+            return false
+          end
+        elsif self[i].nil?
+          return false
+        else
+          return true
+        end
+        i += 1
+      end
+    else
+      return false
+    end
+    true
+  end
+
+  def my_any?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield self[i]
+            return true
+          end
+        else
+          return true
+        end
+        i += 1
+      end
+    else
+      j = 0
+      while j < self.length
+        if args[0] == self[j]
+          return true
+        end
+
+        j += 1
+      end
+
+    end
+    false
+  end
+
+  def my_none?(*args)
+    if args.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield self[i]
+            return false
+          end
+        elsif self[i]
+          return false
+        end
+        i += 1
+      end
+    else
+      j = 0
+      while j < self.length
+        if self[j] == args[0]
+          return false
+        else
+          true
+        end
+
+        j += 1
+      end
+    end
+    true
+  end
+
+  def my_count(*num)
+    count = 0
+    if num.length.zero?
+      i = 0
+      while i < self.length
+        if block_given?
+          if yield self[i]
+            count += 1
+          end
+        else
+          return self.length
+        end
+        i += 1
+      end
+    elsif num.length == 1
+      i = 0
+      while i < self.length
+        if num[0] == self[i]
+          count += 1
+        end
+        i += 1
+      end
+    end
+    count
+  end
+
+  def my_map(my_proc = nil)
+    result = []
+    i = 0
+    while i < self.length
+      if block_given?
+        result.push yield self[i]
+      else
+        result.push my_proc.call(self[i])
+      end
+      i += 1
+    end
+    result
+  end
+
+  def my_inject(my_proc = nil)
+    reduce = nil
+    first_element = self.shift
+    self.unshift(first_element)
+    i = 1
+    while i < self.length
+      if block_given?
+        block_value = yield first_element, self[i]
+        reduce = block_value
+        first_element = reduce
+      elsif my_proc == :+
+        proc_symbol = first_element + self[i]
+        reduce = proc_symbol
+        first_element = reduce
+      elsif my_proc == :*
+        proc_symbol = first_element * self[i]
+        reduce = proc_symbol
+        first_element = reduce
+      elsif my_proc == :-
+        proc_symbol = first_element - self[i]
+        reduce = proc_symbol
+        first_element = reduce
+      elsif my_proc == :/
+        proc_symbol = first_element / self[i]
+        reduce = proc_symbol
+        first_element = reduce
+      end
+      i += 1
+    end
+    reduce
+  end
```

Comment:
- [ ] my_inject raises a "LocalJumpError" when no block or argument is given

- [ ] when a symbol is specified without an initail value, it combines each element of the collection by applying the symbol as a named method in a range
(range.my_inject(:*)) should be equal range.inject(:*)

- [ ] when a symbol is specified with an initail value, it combines each element of the collection by applying the symbol as a named method
expect(array.my_inject(20, :*)) should be equal array.inject(20, :*)

- [ ] when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method in a range
expect(range.my_inject(2, :*)) should be equal range.inject(2, :*)